### ISSUE TITLE
When running Barrier as 'client' on both Mac and Linux, it will always get stuck on "Barrier is starting" regardless of configuration.

### ISSUE THREAD

**Issue by connervieira** [debauchee/barrier#813]
### Operating Systems ###

Server: MacOS Catalina/Pop!_OS 20.4

Client: MacOS Catalina/Pop!_OS 20.4

### Barrier Version ###
Mac: 2.3.3 (Latest for platform)
Linux: 2.3.2 (Latest for platform)

### Steps to reproduce bug ###
When starting as a 'client', Barrier always gets stuck on "Barrier is starting". This happens on both Mac and Linux, and both with manual and automatic set up.
1. Configure one computer to run as server
2. On the other computer, select "Client"
3. Preset "Start"
4. Observe that Barrier gets stuck on "Barrier is starting"

I should note that on first launch, the message that the other machine is attempting to pair shows up, but it still gets stuck on "Barrier is starting"
When running Barrier in server mode, the "Barrier is running" message shows up just as expected.

### Other info ###

* When did the problem start to occur? Ever since I installed Barrier on both platforms (I tried reinstalling)
* Is there a way to work around it? No
* Does this bug prevent you from using Barrier entirely? Yes

**Comment by himdel** [comment]
Did you allow the port 24800 on the server firewall? (And did it work? For example most linux docs mention `firewall-cmd`, which is useless on my linux machine.)

**Comment by connervieira** [comment]
I tried allowing port 24800 on both the server and client, and it still fails. The Mac was the client, and the Linux machine was the server. When I clicked "Start", the Mac showed an error that "Barrier quit unexpectedly", but the GUI was still open and responding. Eventually, the Mac switched from "Barrier is starting" to "Barrier is started", but it still didn't work, which I assume is due to the crash reported by MacOS
